crates/state/src/state_op.rs

```diff
@@ -289,45 +292,41 @@ impl StateCache {
-            }
-
-            /// remove matured block from maturation entry
-            pub fn mature_l1_block(&mut self, idx: u64) {
-                let operators: Vec<_> = self.state().operator_table().indices().collect();
-                let deposits = self.new_state.exec_env_state.pending_deposits_mut();
-                let mqueue = &mut self.new_state.l1_state.maturation_queue;
-
-                // Checks.
-                assert!(mqueue.len() > 1); // make sure we'll still have blocks in the queue
-                let front_idx = mqueue.front_idx().unwrap();
-                assert_eq!(front_idx, idx);
-
-                // Actually take the block out so we can do something with it.
-                let matured_block = mqueue.pop_front().unwrap();
-
-                // TODO add it to the MMR so we can reference it in the future
-                let (header_record, deposit_txs, _) = matured_block.into_parts();
-                for op in deposit_txs.iter().flat_map(|tx| tx.tx().protocol_ops()) {
-                    if let ProtocolOperation::Deposit(deposit_info) = op {
-                        trace!("we got some deposit_txs");
-                        let amt = deposit_info.amt;
-                        let deposit_intent = DepositIntent::new(amt, &deposit_info.address);
-                        deposits.push_back(deposit_intent);
-                        self.new_state
-                            .deposits_table
-                            .add_deposits(&deposit_info.outpoint, &operators, amt)
-                    }
-                }
-
-                self.state_mut().l1_state.safe_block = header_record;
-            }
-    */
+    /// Updates the deposit state to executing.
+    pub fn mark_deposit_executing(&mut self, winfo: &WithdrawalFulfillmentInfo) {
+        let deposit_ent = self
+            .state_mut()
+            .deposits_table_mut()
+            .get_deposit_mut(winfo.deposit_idx)
+            .expect("stateop: missing deposit idx");
+
+        if !matches!(deposit_ent.deposit_state(), DepositState::Dispatched(state) if state.assignee() == winfo.operator_idx)
+        {
+            panic!("stateop: unexpected deposit state");
+        }
+
+        deposit_ent.set_state(DepositState::Fulfilled(FulfilledState::new(
+            winfo.operator_idx,
+            winfo.amt,
+            winfo.txid,
+        )));
+    }
+
+    // Updates the deposit state as Executed.
+    pub fn mark_deposit_spent(&mut self, deposit_idx: u32) {
+        let deposit_ent = self
+            .state_mut()
+            .deposits_table_mut()
+            .get_deposit_mut(deposit_idx)
+            .expect("stateop: missing deposit idx");
+
+        if !matches!(deposit_ent.deposit_state(), DepositState::Fulfilled(_)) {
+            // TODO: handle this better after TN1 bridge is integrated
+            warn!("stateop: deposit spent at unexpected state");
+        }
+
+        deposit_ent.set_state(DepositState::Reimbursed);
+    }
```

Comment:
We're shifting towards simplifying this interface.  These checks can be done outside `StateCache`.  We can just have a `.set_deposit_state` fn.

Reply:
Moving on, doesn't matter.